### ISSUE TITLE
Remove unused GO_REFs

### ISSUE THREAD

**Issue by pgaudet** [geneontology/go-site#1329]
Hello, 

I cannot find those references either in the ontology or in annotations. Can we delete them ? 

GO_REF ID | Description
-- | --
GO_REF:0000006 | Gene Ontology annotation by the MGI   curatorial staff, Mouse Locus Catalog
GO_REF:0000010 | Gene Ontology annotation by the MGI curatorial staff, mouse gene   nomenclature
GO_REF:0000018 | dictyBase 'Inferred from Electronic Annotation (BLAST method)'
GO_REF:0000019 | Automatic transfer of experimentally verified manual GO annotation data   to orthologs using Ense
GO_REF:0000023 | Gene Ontology annotation based on UniProtKB Subcellular Location   vocabulary mapping.
GO_REF:0000026 | Improving the representation of muscle biology in the biological process   and cellular component
GO_REF:0000029 | Gene Ontology annotation based on information extracted from curated   UniProtKB entries
GO_REF:0000030 | Portable Annotation Rules
GO_REF:0000031 | NIAID Cell Ontology Workshop
GO_REF:0000032 | Inference of Biological Process annotations from inter-ontology links
GO_REF:0000033 | (Obsolete) Annotation inferences using phylogenetic trees
GO_REF:0000035 | Automatic transfer of experimentally verified manual GO annotation data   to plant orthologs usin
~GO_REF:0000042~ | Gene Ontology annotation through association of InterPro records with GO   terms, accompanied by
~GO_REF:0000043~ | Gene Ontology annotation based on UniProtKB/Swiss-Prot keyword mapping,   accompanied by conserva
~GO_REF:0000044~ | Gene Ontology annotation based on UniProtKB/Swiss-Prot Subcellular   Location vocabulary mapping,
~GO_REF:0000045~ | Gene Ontology annotation based on UniProtKB/TrEMBL entries keyword   mapping, accompanied by cons
~GO_REF:0000046~ | Gene Ontology annotation based on UniProtKB/TrEMBL Subcellular Location   vocabulary mapping, acc
~GO_REF:0000047~ | Gene Ontology annotation based on absence of key sequence residues. 
GO_REF:0000048 | TIGR's Eukaryotic Manual Gene Ontology Assignment Method
GO_REF:0000049 | Automatic transfer of experimentally verified manual GO annotation data   to fungal orthologs usi
GO_REF:0000053 | Automatic classification of GO using the ELK reasoner
GO_REF:0000055 | Gene Ontology Cellular Component annotation based on cellular   fractionation.
GO_REF:0000056 | Taxon constraints to detect inconsistencies in annotation and ontology   structure.
GO_REF:0000057 | Gene Ontology annotations inferred by curators' judgment using   experimental data and prior know
GO_REF:0000077 | Representation of transport of a cellular component as biological process   in the Gene Ontology
GO_REF:0000082 | Representation of plant maturation as biological process in the Gene   Ontology
GO_REF:0000098 | Gene Ontology annotation based on research conference abstracts
GO_REF:0000099 | Gene Ontology annotation based on DNA/RNA sequence records
GO_REF:0000100 | Gene Ontology annotation by SEA-PHAGE biocurators
GO_REF:0000102 | Representation of cellular component binding as molecular functions in   the Gene Ontology
GO_REF:0000103 | Representation of cellular component organization as biological process   in the Gene Ontology
GO_REF:0000106 | Gene Ontology annotation based on protein sequence records.
GO_REF:0000112 | Gene Ontology annotation by CACAO biocurators

@hdrabkin @pfey @sandyl27 @alexsign 

Do you need any of those references ? 

Thanks, Pascale

**Comment by kltm** [comment]
@pgaudet Don't we treat GO refs as we do ontology entities and not delete them?

**Comment by hdrabkin** [comment]
@ukemi would GO_REF:0000010 before the RCA (Riken) annotations?

**Comment by pgaudet** [comment]
@kltm Sure - I will change the status to 'is_obsolete = true'

**Comment by pgaudet** [comment]
However can we filter obsolete go_refs from our products ? For example GO_REF:0000033 should not be exported: http://release.geneontology.org/2020-01-01/metadata/gorefs/index.html

Thanks, Pascale

**Comment by kltm** [comment]
If there's no longer a landing spot for it, that would miss the point have having the obsolete and notice--just like an ontology term, we don't want to be taking these things "away", just marking them and adding commentary.

The current best landing site would be https://github.com/geneontology/go-site/blob/master/metadata/gorefs/README.md
https://github.com/geneontology/go-site/blob/master/metadata/gorefs/README.md#goref0000010
(unfortunate that the markdown is a little wonky there)

**Comment by pgaudet** [comment]
@alexsign Are you using any one of those GO_REFs?

**Comment by pgaudet** [comment]
GO_REF:0000045, GO_REF:0000046, GO_REF:0000047

Used in UniProt

**Comment by alexsign** [comment]
@pgaudet yes, I do for three of them. see bellow with the number of annotations
GO_REF:0000042	130
GO_REF:0000045	66
GO_REF:0000046	1

**Comment by pgaudet** [comment]
Could 
@alexsign 
Perhaps those should be merged ?? 

GO_REF:0000042 InterPro -> GO_REF:0000002
GO_REF:0000045  SPKW mappings ->  GO_REF:0000004
GO_REF:0000046 -> SubCell -> GO_REF:0000023

What do you think ?

**Comment by alexsign** [comment]
@pgaudet sounds reasonable, I'll let you know when it's remapped, but you can go ahead and remove them now

**Comment by pgaudet** [comment]
thanks !

**Comment by pgaudet** [comment]
What about the others? 
* Is GO_REF:0000042 InterPro2GO (conservatively modified by UniProt) the same as 
GO_REF:0000002 InterPro2GO ? 

* Do we need 
GO_REF:0000037 UniProt Keywords2GO (UniProtKB/Swiss-Prot entries)
GO_REF:0000038 UniProt Keywords2GO (UniProtKB/TrEMBL entries)
in addition to GO_REF:0000043 UniProt Keywords2GO (UniProtKB/Swiss-Prot entries, conservatively modified by UniProt) ? 

* Do we need 
GO_REF:0000039 UniProt Subcellular Location2GO (UniProtKB/Swiss-Prot entries)
GO_REF:0000040 UniProt Subcellular Location2GO (UniProtKB/TrEMBL entries)
in addition to GO_REF:0000044 Gene Ontology annotation based on UniProtKB/Swiss-Prot Subcellular Location vocabulary mapping, accompanied by conservative changes to GO terms applied by UniProt. 

? 

If you don't need them all, let me know which one you'd pick for each group. 

Thanks, Pascale
Let me know whi

**Comment by alexsign** [comment]
@pgaudet the 3 GO_REFs that I posted are the only GO_REFs in the GOA database from the full list of GO_REFs above. It looks like we either stopped or never used GO_REFs 37-40 in GOA.

**Comment by alexsign** [comment]
@pgaudet here is the annotation data for 37-40:
GO_REF:0000038	283932240
GO_REF:0000039	426283
GO_REF:0000040	23391318
GO_REF:0000037	2366409

**Comment by pgaudet** [comment]
ok thanks !

**Comment by pgaudet** [comment]
Hi @alexsign 
@sylvainpoux agrees with the merge: 

* GO_REF:0000037 UniProt Keywords2GO (UniProtKB/Swiss-Prot entries)
*  GO_REF:0000038 UniProt Keywords2GO (UniProtKB/TrEMBL entries)
* '''into'''   in addition to GO_REF:0000043 UniProt Keywords2GO (UniProtKB/Swiss-Prot entries, conservatively modified by UniProt)

* GO_REF:0000039 UniProt Subcellular Location2GO (UniProtKB/Swiss-Prot entries)
*  GO_REF:0000040 UniProt Subcellular Location2GO (UniProtKB/TrEMBL entries)
* '''into''' REF:0000044 Gene Ontology annotation based on UniProtKB/Swiss-Prot Subcellular Location vocabulary mapping, accompanied by conservative changes to GO terms applied by UniProt.


Let me know if I can remove references 37, 38, 39 and 40 or if you need to merge them at your end first.

Thanks, Pascale

**Comment by alexsign** [comment]
@pgaudet please hold on for now. I'll need do all needed changes in the database and unload data to QuickGO first.

**Comment by pgaudet** [comment]
Sounds good.

**Comment by alexsign** [comment]
@pgaudet it's all good now in GOA and QuickGO. Please go ahead and absolute GO_REFs discussed above. Thanks.

**Comment by suzialeksander** [comment]
Looks like trying to annotate a yeast protein to "GO:0045053 protein retention in Golgi apparatus" is getting 

> Annotation failed taxon constraint GO:0032507 (maintenance of protein location in cell) and its descendants should ONLY be used with gene products from the taxonomic group cellular organisms

is this something on our (GO's) end or P2GO?

**Comment by pgaudet** [comment]
Obsolete GO_REF:0000004 - has xrefs to SGD, ZFIN and MGI 

- not used by SGD: http://release.geneontology.org/2021-09-01/products/annotations/sgd_valid.gaf.gz

Used by ZFIN and MGI
- ZDB-PUB-020723-1 used by ZFIN: http://release.geneontology.org/2021-09-01/products/annotations/zfin_valid.gaf.gz 
- MGI: MGI:MGI:1354194|GO_REF:0000004 http://release.geneontology.org/2021-09-01/products/annotations/mgi_valid.gaf.gz
